crates/cudnn/README.md

```diff
@@ -1,2 +1,84 @@
 # cudnn
 Type safe cuDNN wrapper for the Rust programming language.
+
+## Project status
+The current version of cuDNN targeted by this wrapper is the 8.3.2. You can refer to the official [release notes](https://docs.nvidia.com/deeplearning/cudnn/release-notes/index.html) and to the [support matrix](https://docs.nvidia.com/deeplearning/cudnn/support-matrix/index.html) by NVIDIA.
+
+The legacy API is somewhat complete and it is usable but the backend API is still to be considered a work in progress and its usage is therefore much discouraged. Both APIs are still being developed so expect bugs and reasonable breaking changes whilst using this crate. 
+
+The project is part of the Rust CUDA ecosystem and is actively maintained by [frjnn](https://github.com/frjnn).
+
+## Primer 
+
+Here follows a list of useful concepts that should be taken as a handbook for the users of the crate. This is not intended to be the full documentation, as each wrapped struct, enum and function has its own docs, but rather a quick sum up of the key points of the API. As a matter of fact, for a deeper view, you should refer both to the docs of each item and to the [official ones](https://docs.nvidia.com/deeplearning/cudnn/api/index.html#overview) by NVIDIA. Furthermore, if you are new to cuDNN we strongly suggest reading the [official developer guide](https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#overview).
+
+### Device buffers
+
+This crate is built around [`cust`](https://docs.rs/cust/latest/cust/memory/index.html) own memory routines and transfer functions.
+
+### cuDNN statuses and Result
+
+All cuDNN library functions return their status. This crate uses [`Result`](https://doc.rust-lang.org/std/result/enum.Result.html) to achieve a leaner, idiomatic and easier to manage API.
+
+### cuDNN handles and RAII
+
+The main entry point of the cuDNN library is the `CudnnContext` struct. This handle is tied to a device and it is explicitly passed to every subsequent library function that operates on GPU data. It manages resources allocations both on the host and the device and takes care of the synchronization of all the the cuDNN primitives. 
+
+The handles, and the other cuDNN structs wrapped by this crate, are implementors of the [`Drop`](https://doc.rust-lang.org/std/ops/trait.Drop.html) trait which implicitly calls their destructors on the cuDNN side when they go out of scope. 
+
+cuDNN contexts can be created as shown in the following snippet:
+
+```rust
```

Comment:
cudnn requires u to init first, no? at least cust does, maybe include initialization in this snippet?

Reply:
the snippet runs fine, but I will add the ctx init, moreover many cudnn structs can be initialised without a ctx, although the docs specify otherwise

Reply:
oh wonderful